File: replace_alerts.js
const fs = require('fs');
const path = require('path');

const directory = path.join(__dirname, 'frontend', 'src');

function processDirectory(dir) {
    const files = fs.readdirSync(dir);
    for (const file of files) {
        const fullPath = path.join(dir, file);
        const stat = fs.statSync(fullPath);
        if (stat.isDirectory()) {
            processDirectory(fullPath);
        } else if (file.endsWith('.tsx') || file.endsWith('.ts')) {
            let content = fs.readFileSync(fullPath, 'utf8');
            if (content.includes('alert(')) {
                if (!content.includes("import toast from 'react-hot-toast';")) {
                    const importMatch = content.match(/^import .*?;/gm);
                    if (importMatch) {
                        const lastImport = importMatch[importMatch.length - 1];
                        content = content.replace(lastImport, lastImport + "\nimport toast from 'react-hot-toast';");
                    } else {
                        content = "import toast from 'react-hot-toast';\n" + content;
                    }
                }
                
                content = content.replace(/alert\((['"])(.*?[Ff]ail.*?)['"]\)/g, "toast.error($1$2$1)");
                content = content.replace(/alert\((['"])(.*?[Ee]rror.*?)['"]\)/g, "toast.error($1$2$1)");
                content = content.replace(/alert\((['"])(.*?[Ii]nvalid.*?)['"]\)/g, "toast.error($1$2$1)");
                content = content.replace(/alert\(/g, "toast.success(");

                fs.writeFileSync(fullPath, content, 'utf8');
                console.log(`Updated ${file}`);
            }
        }
    }
}

processDirectory(directory);

File: replace_alerts.py
import os
import re

directory = r"d:\Building Approval Software\frontend\src"

for root, _, files in os.walk(directory):
    for file in files:
        if file.endswith(".tsx") or file.endswith(".ts"):
            filepath = os.path.join(root, file)
            with open(filepath, 'r', encoding='utf-8') as f:
                content = f.read()

            if 'alert(' in content:
                # Add import if not present
                if "import toast from 'react-hot-toast';" not in content:
                    # Find last import
                    imports = [m for m in re.finditer(r'^import .*?;$', content, re.MULTILINE)]
                    if imports:
                        last_import = imports[-1]
                        content = content[:last_import.end()] + "\nimport toast from 'react-hot-toast';" + content[last_import.end():]
                    else:
                        content = "import toast from 'react-hot-toast';\n" + content

                # Replace alerts. Simple heuristic: 
                # alert('Failed to submit enquiry.') -> toast.error('Failed to submit enquiry.')
                # alert('Thank you...') -> toast.success('Thank you...')
                
                # replace alert('Failed or alert("Failed
                content = re.sub(r'alert\(([\'"])(.*?[Ff]ail.*?)[\'"]\)', r'toast.error(\1\2\1)', content)
                content = re.sub(r'alert\(([\'"])(.*?[Ee]rror.*?)[\'"]\)', r'toast.error(\1\2\1)', content)
                content = re.sub(r'alert\(([\'"])(.*?[Ii]nvalid.*?)[\'"]\)', r'toast.error(\1\2\1)', content)
                
                # remaining alerts become toast.success
                content = re.sub(r'alert\(', r'toast.success(', content)

                with open(filepath, 'w', encoding='utf-8') as f:
                    f.write(content)
                print(f"Updated {file}")
